druid/src/win_handler.rs

```diff
@@ -336,6 +389,25 @@ impl<T: Data> Inner<T> {
         }
     }
 
+    fn post_render(&mut self, window_id: WindowId) {
+        // Pre-order traversal, so that children render on top of parents
+        if let Some(win) = self.windows.get_mut(window_id) {
+            // Render self first
+            win.do_post_render();
+
+            // Render children next
+            let mut child_ids = vec![];
```

Comment:
I don't get what this vec is for

Reply:
Probably just me being a novice in Rust, I couldn't manage to write the second `for` loop below without the borrow checker complaining about a mutable reference already in use, so I had to first extract the child IDs then release the mutable reference to `win` I think, then finally do the loop with the `do_post_render()` call. I'm a bit fuzzy on details. Happy to be taught a better way here.

Reply:
Ah I see. No I don't know a way to avoid that, disjoint borrows of collections is a bit of a nightmare. 

It could be written as and maybe a comment saying we clone into a vector to satisfy the borrow checker. 

for child_id in win.children.iter().cloned().collect::<Vec<_>>(){
   ... stuff
}

You could avoid the vec allocation if we limit the number of child windows and put it in a fixed length slice on the stack, or fail over to the vec version when the number goes over some limit.. But probably not worth it....

Reply:
Actually that's a good point:
- Currently the change only support 1 level of parenting depth. I am not sure if we can make things work for a random parenting depth, but the vector is clearly overkill for now.
- Even if that is implemented, it is unlikely we will see deep hierarchies of native windows. So a fast no-alloc handling for a smaller number is probably a good thing to have for 99% of the cases.
- Painting happens a lot more often than parenting, so we could keep the vector around allocated, and only try to resize it, which would be a no-op almost all of the time.